rpc/src/rpc.rs

```diff
@@ -554,51 +592,103 @@ impl JsonRpcRequestProcessor {
             }
         }
 
+        let bank = self.get_bank_with_config(slot_context)?;
+
         let first_confirmed_block_in_epoch = *self
             .get_blocks_with_limit(first_slot_in_epoch, 1, config.commitment)
             .await?
             .first()
             .ok_or(RpcCustomError::BlockNotAvailable {
                 slot: first_slot_in_epoch,
             })?;
+        let partitioned_epoch_reward_enabled_slot = bank
+            .feature_set
+            .activated_slot(&feature_set::enable_partitioned_epoch_reward::id());
+        let partitioned_epoch_reward_enabled = partitioned_epoch_reward_enabled_slot
+            .map(|slot| slot <= first_confirmed_block_in_epoch)
+            .unwrap_or(false);
 
-        let Ok(Some(first_confirmed_block)) = self
-            .get_block(
+        let mut reward_map: HashMap<String, (Reward, Slot)> = {
+            let addresses: Vec<String> =
+                addresses.iter().map(|pubkey| pubkey.to_string()).collect();
+
+            self.get_reward_map(
                 first_confirmed_block_in_epoch,
-                Some(RpcBlockConfig::rewards_with_commitment(config.commitment).into()),
+                &addresses,
+                &|reward_type| -> bool {
+                    reward_type == RewardType::Voting
+                        || (!partitioned_epoch_reward_enabled && reward_type == RewardType::Staking)
```

Comment:
i was thinking we pass in a different closure dependent on feature activation, but i guess since we can't actually remove the pre-activation logic, this is fine.